include/aspect/particle/base_particle.h

```diff
@@ -114,8 +104,17 @@ namespace aspect
          *
          * @return Number of doubles required to represent this particle
          */
-        static unsigned int
-        data_len ();
+        unsigned int
+        data_len () const;
```

Comment:
could we call this `n_properties()` and the next function `set_n_properties()`, or similar? Or does the function include storage for properties *other* than those stored in `val`? (In that case, maybe updating the comment would be useful.)